chain/deals/provider_states.go

```diff
@@ -221,89 +180,31 @@ func (p *Provider) staged(ctx context.Context, deal MinerDeal) (func(*MinerDeal)
 	if err != nil {
 		return nil, xerrors.Errorf("getting unixfs file size: %w", err)
 	}
-	if uint64(size) != deal.Proposal.PieceSize {
-		return nil, xerrors.Errorf("deal.Proposal.PieceSize didn't match unixfs file size")
+	if padreader.PaddedSize(uint64(size)) != deal.Proposal.PieceSize {
+		return nil, xerrors.Errorf("deal.Proposal.PieceSize didn't match padded unixfs file size")
 	}
 
-	pcid, err := cid.Cast(deal.Proposal.PieceRef)
-	if err != nil {
-		return nil, err
-	}
-
-	sectorID, err := p.secst.AddUnixfsPiece(pcid, uf, deal.DealID)
+	sectorID, err := p.secb.AddUnixfsPiece(ctx, deal.Ref, uf, deal.DealID)
 	if err != nil {
 		return nil, xerrors.Errorf("AddPiece failed: %s", err)
 	}
-
 	log.Warnf("New Sector: %d", sectorID)
+
 	return func(deal *MinerDeal) {
 		deal.SectorID = sectorID
 	}, nil
 }
 
 // SEALING
 
-func (p *Provider) waitSealed(ctx context.Context, deal MinerDeal) (sectorbuilder.SectorSealingStatus, error) {
-	status, err := p.secst.WaitSeal(ctx, deal.SectorID)
-	if err != nil {
-		return sectorbuilder.SectorSealingStatus{}, err
-	}
-
-	switch status.State {
-	case sealing_state.Sealed:
-	case sealing_state.Failed:
-		return sectorbuilder.SectorSealingStatus{}, xerrors.Errorf("sealing sector %d for deal %s (ref=%s) failed: %s", deal.SectorID, deal.ProposalCid, deal.Ref, status.SealErrorMsg)
-	case sealing_state.Pending:
-		return sectorbuilder.SectorSealingStatus{}, xerrors.Errorf("sector status was 'pending' after call to WaitSeal (for sector %d)", deal.SectorID)
-	case sealing_state.Sealing:
-		return sectorbuilder.SectorSealingStatus{}, xerrors.Errorf("sector status was 'wait' after call to WaitSeal (for sector %d)", deal.SectorID)
-	default:
-		return sectorbuilder.SectorSealingStatus{}, xerrors.Errorf("unknown SealStatusCode: %d", status.SectorID)
-	}
-
-	return status, nil
-}
-
 func (p *Provider) sealing(ctx context.Context, deal MinerDeal) (func(*MinerDeal), error) {
-	err := p.sendSignedResponse(&Response{
-		State:    api.DealSealing,
-		Proposal: deal.ProposalCid,
-	})
-	if err != nil {
-		log.Warnf("Sending deal response failed: %s", err)
-	}
-
-	if err := p.secst.SealSector(ctx, deal.SectorID); err != nil {
-		return nil, xerrors.Errorf("sealing sector failed: %w", err)
-	}
-
-	_, err = p.waitSealed(ctx, deal)
-	if err != nil {
-		return nil, err
-	}
-	// TODO: Spec doesn't say anything about inclusion proofs anywhere
-	//  Not sure what mechanisms prevents miner from storing data that isn't
-	//  clients' data
+	// TODO: consider waiting for seal to happen
```

Comment:
Are we fine leaving this unhandled for now?

Reply:
Yes, that would be only useful for display purposes right now (will be more useful when we implement graceful restarting later)